### ISSUE TITLE
Image.DetectFormat does not return null on invalid format

### ISSUE THREAD

**Issue by BitSchupser** [SixLabors/ImageSharp#714]
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of ImageSharp
- [ ] I have verified if the problem exist in both `DEBUG` and `RELEASE` mode
- [x] I have searched [open](https://github.com/SixLabors/ImageSharp/issues) and [closed](https://github.com/SixLabors/ImageSharp/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description

When calling `Image.DetectFormat()` with a JPEG byte-array at first and with an invalid array second, the second call returns `ImageFormat.Jpeg` but should return `null`.

### Steps to Reproduce

This unit test fails because of the bug:

```
[TestMethod]
public void Failing_test_because_DetectFormat_does_not_clean_up()
{
	byte[] jpegImage;
	using (var buffer = new MemoryStream())
	{
		using (var image = new Image<Rgba32>(100, 100))
		{
			image.SaveAsJpeg(buffer);
			jpegImage = buffer.ToArray();
		}
	}

	byte[] invalidImage = { 1, 2, 3 };

	Image.DetectFormat(jpegImage).ShouldBe(ImageFormats.Jpeg);
	Image.DetectFormat(invalidImage).ShouldBeNull();
}
```

### Workaround
When manually releasing the resources at MemoryManager it works:

```
[TestMethod]
public void Workaround_with_manually_cleaning_up ()
{
	byte[] jpegImage;
	using (var buffer = new MemoryStream())
	{
		using (var image = new Image<Rgba32>(100, 100))
		{
			image.SaveAsJpeg(buffer);
			jpegImage = buffer.ToArray();
		}
	}

	byte[] invalidImage = { 1, 2, 3 };

	Image.DetectFormat(jpegImage).ShouldBe(ImageFormats.Jpeg);
	SixLabors.ImageSharp.Configuration.Default.MemoryManager.ReleaseRetainedResources();
	Image.DetectFormat(invalidImage).ShouldBeNull();
}
```

### System Configuration

- ImageSharp version: 1.0.0-beta0004
- Other ImageSharp packages and versions: none
- Environment (Operating system, version and so on): Windows 10 Enterprise, Version 1703 Build 15063.1324
- .NET Framework version: dotnet core 2.1.401

**Comment by JimBobSquarePants** [comment]
[ ]  I have verified that I am running the latest version of ImageSharp

**Comment by JimBobSquarePants** [comment]
Fix confirmed and PR Opened #715

**Comment by mocanSergiu666** [comment]
I have an opposite situation, I always get null when trying to detect the image format.

private bool IsAvatarInvalid(IFormFile avatar) {
    Stream avatarStream = new MemoryStream();

    avatar.CopyTo(avatarStream);

    IImageFormat imageFormat = Image.DetectFormat(avatarStream);

    return imageFormat.Name;
}

imageFormat is always null :(
